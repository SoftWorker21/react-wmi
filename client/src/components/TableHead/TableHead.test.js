import { render, screen } from "@testing-library/react";
import { TableHead } from "./index";
import data from "../../honda_wmi.json";

const props = {
  carData: data,
  currentItems: data,
  setCurrentItems: jest.fn(),
};

test("render CreatedOn react link", () => {
  render(<TableHead {...props} />);
  const element = screen.getByText(/CreatedOn/i);
  expect(element).toBeInTheDocument();
});

test("render VehicleType react link", () => {
  render(<TableHead {...props} />);
  const element = screen.getByText(/VehicleType/i);
  expect(element).toBeInTheDocument();
});

test("render Name react link", () => {
  render(<TableHead {...props} />);
  const element = screen.getByText(/Name/i);
  expect(element).toBeInTheDocument();
});

test("render Header snapshot", () => {
  const tree = render(<TableHead {...props} />);
  expect(tree).toMatchSnapshot();
});
