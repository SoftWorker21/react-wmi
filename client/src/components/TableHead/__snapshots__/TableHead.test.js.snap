// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Header snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <thead>
        <tr>
          <th
            class="Name"
          >
            <div
              class="row-title"
            >
              <p
                class="row-title--text"
              >
                Name
              </p>
              <p
                class="down-arrow"
              >
                ⌃
              </p>
            </div>
          </th>
          <th
            class="WMI"
          >
            <div
              class="row-title"
            >
              <p
                class="row-title--text"
              >
                WMI
              </p>
              <p
                class="down-arrow"
              >
                ⌃
              </p>
            </div>
          </th>
          <th
            class="Country"
          >
            <div
              class="row-title"
            >
              <p
                class="row-title--text"
              >
                Country
              </p>
              <p
                class="down-arrow"
              >
                ⌃
              </p>
            </div>
          </th>
          <th
            class="CreatedOn"
          >
            <div
              class="row-title"
            >
              <p
                class="row-title--text"
              >
                CreatedOn
              </p>
              <p
                class="down-arrow"
              >
                ⌃
              </p>
            </div>
          </th>
          <th
            class="VehicleType"
          >
            <div
              class="row-title"
            >
              <p
                class="row-title--text"
              >
                VehicleType
              </p>
              <p
                class="down-arrow"
              >
                ⌃
              </p>
            </div>
          </th>
        </tr>
      </thead>
    </div>
  </body>,
  "container": <div>
    <thead>
      <tr>
        <th
          class="Name"
        >
          <div
            class="row-title"
          >
            <p
              class="row-title--text"
            >
              Name
            </p>
            <p
              class="down-arrow"
            >
              ⌃
            </p>
          </div>
        </th>
        <th
          class="WMI"
        >
          <div
            class="row-title"
          >
            <p
              class="row-title--text"
            >
              WMI
            </p>
            <p
              class="down-arrow"
            >
              ⌃
            </p>
          </div>
        </th>
        <th
          class="Country"
        >
          <div
            class="row-title"
          >
            <p
              class="row-title--text"
            >
              Country
            </p>
            <p
              class="down-arrow"
            >
              ⌃
            </p>
          </div>
        </th>
        <th
          class="CreatedOn"
        >
          <div
            class="row-title"
          >
            <p
              class="row-title--text"
            >
              CreatedOn
            </p>
            <p
              class="down-arrow"
            >
              ⌃
            </p>
          </div>
        </th>
        <th
          class="VehicleType"
        >
          <div
            class="row-title"
          >
            <p
              class="row-title--text"
            >
              VehicleType
            </p>
            <p
              class="down-arrow"
            >
              ⌃
            </p>
          </div>
        </th>
      </tr>
    </thead>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
